src/components/infoWindow.vue

```diff
@@ -44,37 +40,129 @@ const props = {
   }
 }
 
+const props = {
+  options: {
+    type: Object,
+    default(){return {};}
+  },
+  content: {
+    default(){return null;}
+  },
+  opened: {
+    type: Boolean,
+    default(){return true;}
+  },
+  position: {
+    type: Object,
+  },
+  zIndex: {
+    type: Number,
+  }
+}
+
 const events = [
   'domready',
   'closeclick'
 ]
 
-
+const getLocalField = function (self, field){
+  return (typeof self.$options.propsData[field] !== 'undefined')?self[field]:self.infoWindowObj[field];
+};
+const setLocalField = function (self, field, value){
+  self.infoWindowObj[field] = value;
+  self.$emit(field.replace(/([a-z](?=[A-Z]))/g, '$1-').toLowerCase()+'_changed', value);
+  self.$nextTick(function (){
+    self.infoWindowObj[field] = getLocalField(self, field);
+  });
+};
 export default MapComponent.extend({
-  replace: false,
   props: props,
-  
+  data(){
+    return {
+      infoWindowObj: {
+        options:{},
+        content:null,
+        opened:null,
+        position:{},
+        zIndex:null
+      }
+    };
+  },
+  computed:{
+    local_options:{
+      get(){
+        return getLocalField(this, 'options');
+      },
+      set(value){
+        setLocalField(this, 'options', value);
+      }
+    },
+    local_content:{
+      get(){
+        return getLocalField(this, 'content');
+      },
+      set(value){
+        setLocalField(this, 'content', value);
+      }
+    },
+    local_opened:{
+      get(){
+        return getLocalField(this, 'opened');
+      },
+      set(value){
+        this.infoWindowObj.opened = value;
+        this.$emit('opened_changed', value);
+        this.$nextTick(function (){
```

Comment:
Shouldn't you be using a lexical this here?

```js
this.$nextTick( () => {
```

to preserve context?

I may be completely wrong though depending on how $nextTick works internally